BUILD_WORKFLOWS.md

```diff
@@ -0,0 +1,104 @@
+# 🧰 Using the Workflow Builder Image
+
+This container image provides a self-contained environment for building and deploying SonataFlow projects. It wraps a `build.sh` script that simplifies the process from manifest generation to deployment.
+The script can be used directly from Linux/MacOS instead of using the container image, given all of the tools avaiable.
+
+---
+
+## ✅ What the Image Does
+
+When you run the container, it invokes the `build.sh` script, which performs the following:
+
+1. **Generates SonataFlow manifests** using the [`kn-workflow`](https://github.com/kiegroup/kogito-tooling) plugin (v1.35.0 required).
+2. **Builds the workflow image** using either Podman or Docker.
+3. **Optionally deploys** the workflow:
+   - Pushes the built image to the container registry.
+   - Applies the generated manifests to the current Kubernetes cluster using `kubectl`.
+
+---
+
+## 📌 Script Usage
+
+```text
+Usage:
+/app/scripts/build.sh [flags]
+```
+
+### Flags:
+
+| Flag | Description |
+|------|-------------|
+| `-i`, `--image=<string>` | **(Required)** Full image name to build and optionally push (e.g., `quay.io/org/app:tag`). |
+| `-b`, `--builder-image=<string>` | Override the builder image used during the workflow image build. |
+| `-r`, `--runtime-image=<string>` | Override the base image for running the workflow. |
+| `-n`, `--namespace=<string>` | Target Kubernetes namespace. Defaults to current context namespace. |
+| `-m`, `--manifests-directory=<string>` | Directory for outputting the generated manifests. Defaults to `./manifests`. |
+| `-w`, `--workflow-directory=<string>` | Directory containing the workflow's `src/`. Defaults to current directory. |
+| `-P`, `--no-persistence` | Disables persistence configuration in SonataFlow CR. |
+| `--push` | Pushes the workflow image to the specified registry. |
+| `--deploy` | Applies the generated manifests to the cluster. |
+| `-h`, `--help` | Displays usage help. |
+
+---
+
+## 🚀 Running the Container
+
+Here’s a typical command used to build and optionally push the workflow image:
+
+```bash
+podman run --rm \
+  --network=host \
+  --privileged \
+  -v $HOME/.config/containers/auth.json:/root/.config/containers/auth.json:ro \
+  -v $(pwd)/05_software_template_hello_world/workflow:/workspace \
+  quay.io/orchestrator/orchestrator-workflow-builder:1.35 \
+  -i quay.io/masayag/builder-test:test \
```

Comment:
I would parametrized the namespace  => `quay.io/${REGISTRY_NAMESPACE}/builder-test:test` and I would use the same in the `-n` flag value when running the build script

Reply:
ok then I would at least replace `masayag` by `<replaceme>` or something like that so it stands out that it must be changed and ppl will not use your repo because the read it too quickly (and will likely have an error)

Reply:
ah, you did something similar with `TARGET_WORKFLOW_IMAGE`, cool!